### ISSUE TITLE
pcpp cannot handle our code: What's the best path forward?

### ISSUE THREAD

**Issue by mathomp4** [Fortran-FOSS-Programmers/ford#600]
In Ford 7, `pcpp` is now the default preprocessor. We've found that with MAPL this does not work, see https://github.com/GEOS-ESM/MAPL/actions/runs/6933072181/job/18858418856 for an example.

We can duplicate this locally on a mac if `pcpp` is installed. Now, MAPL is pretty complex Fortran with *lots* of includes to gFTL which is a quite complex set of macros, perhaps `pcpp` does not like it? 

We can "fix" this issue by adding:
```markdown
preprocessor: cpp -traditional-cpp -E
```
in our control file and all is well.

So, what is the right path forward? This seems a bug in `pcpp` but I'm not sure internally how `pcpp` is called by Ford. @ZedThree Can you tell me how I could work on getting a reproducer to the `pcpp` folks to show them they have an issue?

**Comment by mathomp4** [comment]
Okay. I think I see the issue:
```
❯ rg 'procedure :: capacity'
file_from_cpp12
638:         procedure :: capacity => ti_capacity
2086:         procedure :: capacity => tt_capacity

file_from_pcpp
351:         procedure :: capacity => ti_ capacity
1456:         procedure :: capacity => tt_ capacity
```
for some reason, pcpp is putting in an extra space in a *LOT* of calls. So it's getting `ti_ capacity` instead of `ti_capacity`.

**Comment by mathomp4** [comment]
Okay. Issue filed at pcpp, https://github.com/ned14/pcpp/issues/84, but I'm guessing the fact we seem to rely on `-traditional-cpp` means we might be out of luck. Luckily, not too hard to get GNU cpp on most systems.

**Comment by ZedThree** [comment]
Ah, that's a pain! This is probably worth turning into a FAQ, I imagine it will come up again

**Comment by mathomp4** [comment]
Yeah. And it looks like it's a wontfix for the pcpp folks (as expected). As I said in that issue, the "right" thing is to wait for the Fortran 202Y Standard Preprocessor and we can make a `pfpp` at that point.

**Comment by mathomp4** [comment]
Also, as an aside, I think the readthedocs here still points to the Ford 6 info. That's why I was a bit confused about the pcpp stuff.
